pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,237 @@
+			logger.Infof("Creating Issuer %s/%s.", namespace, issuerName)
+			issuer = &api.Issuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      issuerName,
+					Namespace: namespace,
+				},
+				Spec: api.IssuerSpec{
+					IssuerConfig: api.IssuerConfig{
+						SelfSigned: &api.SelfSignedIssuer{},
+					},
+				},
+			}
+			issuers, _ := clusterdContext.Admission.Issuers(namespace).Create(ctx, issuer, metav1.CreateOptions{})
+			sleepTime := 1
+			attempts := 10
+			for i := 0; i < attempts; i++ {
+				for j := range issuers.Status.Conditions {
+					if issuers.Status.Conditions[j].Status == v1.ConditionStatus(api.IssuerConditionReady) {
+						return issuers, nil
+					}
+					logger.Debugf("webhook config %q status=%+v", issuerName, issuers.Status.Conditions[j].Status)
+				}
+				time.Sleep(time.Duration(sleepTime) * time.Second)
+				logger.Infof("waiting for webhook config %q to be in ready status", issuerName)
+			}
+			return issuers, nil
+		}
+
+	}
+
+	return issuer, nil
+}
+
+func createCertificate(ctx context.Context, clusterdContext *clusterd.Context, issuer *api.Issuer) error {
+	logger.Infof("Fetching certificate %s/%s.", namespace, certificateName)
+	_, err := clusterdContext.Admission.Certificates(namespace).Get(ctx, certificateName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Infof("Creating certificate %s/%s.", namespace, certificateName)
+			certificate := &api.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      certificateName,
+					Namespace: namespace,
+				},
+				Spec: api.CertificateSpec{
+					DNSNames: []string{
+						admissionControllerAppName,
+						fmt.Sprintf("%s.%s.svc", admissionControllerAppName, namespace),
+						fmt.Sprintf("%s.%s.svc.cluster.local", admissionControllerAppName, namespace)},
+					IssuerRef: v1.ObjectReference{
+						Name: issuer.Name,
+						Kind: "Issuer",
+					},
+					SecretName: admissionControllerAppName,
+				},
+			}
+			cert, _ := clusterdContext.Admission.Certificates(namespace).Create(ctx, certificate, metav1.CreateOptions{})
+			sleepTime := 1
+			attempts := 10
+			for i := 0; i < attempts; i++ {
+				for j := range cert.Status.Conditions {
+					if cert.Status.Conditions[j].Status == v1.ConditionStatus(api.CertificateConditionReady) {
+						return nil
+					}
+					logger.Debugf("webhook config %q status=%+v", certificateName, cert.Status.Conditions[j].Status)
+				}
+				logger.Infof("waiting for webhook config %q to be in ready status", certificateName)
+				time.Sleep(time.Duration(sleepTime) * time.Second)
+			}
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func createWebhook(ctx context.Context, clusterdContext *clusterd.Context) error {
+	sideEffects := admv1.SideEffectClassNone
+	var timeout int32 = 5
+	serviceCephClusterPath := "/validate-ceph-rook-io-v1-cephcluster"
+	serviceCephBlockPoolPath := "/validate-ceph-rook-io-v1-cephblockpool"
+	serviceCephObjectStorePath := "/validate-ceph-rook-io-v1-cephobjectstore"
```

Comment:
In the integration tests can we test creating an invalid cluster, pool, and object store CR that will each be rejected by their admission controller?

Reply:
Let's discuss how we can integrate with current tests or I think we can make in follow-up PR?

Reply:
follow up sounds fine, could you open an issue to track?

Reply:
crated issue https://github.com/rook/rook/issues/10024